modules/internal/LocaleModelHelpMem.chpl

```diff
@@ -67,6 +67,26 @@ module LocaleModelHelpMem {
     return chpl_mem_alloc(size.safeCast(c_size_t), md + chpl_memhook_md_num());
   }
 
+  import Allocators;
+  pragma "allocator"
+  pragma "llvm return noalias"
+  pragma "always propagate line file info"
+  proc chpl_here_alloc_with_allocator(size:int(64),
+                                      md:chpl_mem_descInt_t,
+                                      ref allocator: record): c_ptr(void) {
```

Comment:
They are both necessary because I maintained the ability to make an allocator either a class or a record. And they can't be a single generic definition because the `record` case is `ref` and the `class` case `const ref`, and the function cannot be overloaded on intent alone.